### ISSUE TITLE
Permissions for modules (e.g. administration, help desk) via LDAP groups don't work anymore in recent snapshots

### ISSUE THREAD

**Issue by msoltyspl** [pwm-project/pwm#583]
I observed it in at least last 3 snapshots. If you configure the settings as e.g.

![image](https://user-images.githubusercontent.com/3541095/100281444-365c7580-2f6a-11eb-950b-598a2d1792c1.png)

then those relationships will not be recognized after the relevant users log into pwm interface, for example - if "Michał Sołtys" logs in, no administrative options are available:

![image](https://user-images.githubusercontent.com/3541095/100262802-f20eac80-2f4c-11eb-9111-e92ca0488c07.png)

Trace log attached:
[PWM.issue3.log](https://github.com/pwm-project/pwm/files/5598377/PWM.issue3.log)

Interesting part:

```
2020-11-25T18:34:14Z, TRACE, impl.AbstractChaiEntry, ruleMap = {chai.pwrule.repeat.max=0, chai.pwrule.changeMessage=, chai.pwrule.numeric.allow=true, chai.pwrule.upper.min=0, chai.pwrule.allowUserChange=true, chai.pwrule.disallowedValues=, chai.pwrule.challengeResponseEnabled=false, chai.pwrule.length.max=0, chai.pwrule.nonalpha.allow=true, chai.pwrule.nonalpha.max=0, chai.pwrule.allowAdminChange=true, chai.pwrule.disallowedAttributes=, chai.pwrule.uniqueRequired=true, chai.pwrule.ADComplexity2008=false, chai.pwrule.sequentialRepeat.max=0, chai.pwrule.lower.min=0, chai.pwrule.ADComplexity=false, chai.pwrule.unique.max=0, chai.pwrule.special.max=0, chai.pwrule.enforceAtLogin=false, chai.pwrule.nonalpha.min=0, chai.pwrule.numeric.allowLast=true, chai.pwrule.policyEnabled=true, chai.pwrule.numeric.allowFirst=true, chai.pwrule.special.allow=true, chai.pwrule.expirationInterval=15552000, chai.pwrule.special.min=0, chai.pwrule.lower.max=0, chai.pwrule.numeric.max=0, chai.pwrule.novellComplexity=, chai.pwrule.ADComplexityMaxViolation=false, chai.pwrule.upper.max=0, chai.pwrule.numeric.min=0, chai.pwrule.unique.min=0, chai.pwrule.special.allowFirst=true, chai.pwrule.length.min=12, chai.pwrule.special.allowLast=true, chai.pwrule.caseSensitive=true, chai.pwrule.lifetime.minimimum=0}
2020-11-25T18:34:14Z, TRACE, ldap.UserInfoReader, {hIv2C} read password expiration time: 2021-05-24T12:01:45Z, 179d:18h:27m from now [172.17.0.1]
2020-11-25T18:34:14Z, TRACE, permission.UserPermissionUtility, {hIv2C} begin check for ldapQuery match for cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) using queryMatch: (&(objectClass=toukPerson)(toukAccountActive=TRUE)) [172.17.0.1]
2020-11-25T18:34:14Z, TRACE, permission.UserPermissionUtility, {hIv2C} checking ldap to see if cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) matches '(&(objectClass=toukPerson)(toukAccountActive=TRUE))' [172.17.0.1]
2020-11-25T18:34:14Z, DEBUG, permission.UserPermissionUtility, {hIv2C} user cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) is a match for permission 'UserPermission(type=ldapQuery, ldapProfileID=default, ldapQuery=(&(objectClass=toukPerson)(toukAccountActive=TRUE)), ldapBase=ou=People,dc=touk,dc=pl)' (1ms) [172.17.0.1]
2020-11-25T18:34:14Z, DEBUG, ldap.UserInfoReader, {hIv2C} assigned ChangePassword profileID "default" to cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) [172.17.0.1]
2020-11-25T18:34:14Z, DEBUG, permission.UserPermissionUtility, {hIv2C} user cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) is a match for permission 'UserPermission(type=ldapAllUsers, ldapProfileID=all, ldapQuery=null, ldapBase=null)' (0ms) [172.17.0.1]
2020-11-25T18:34:14Z, DEBUG, ldap.UserInfoReader, {hIv2C} assigned AccountInformation profileID "default" to cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) [172.17.0.1]
2020-11-25T18:34:14Z, TRACE, permission.LdapGroupTypeHelper, {hIv2C} begin check for ldapGroup match for cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) using queryMatch: null [172.17.0.1]
2020-11-25T18:34:14Z, TRACE, permission.LdapGroupTypeHelper, {hIv2C} missing groupDN value, skipping check [172.17.0.1]
2020-11-25T18:34:14Z, DEBUG, permission.LdapGroupTypeHelper, {hIv2C} user cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) is not a match for group 'null' (0ms) [172.17.0.1]
2020-11-25T18:34:14Z, DEBUG, permission.UserPermissionUtility, {hIv2C} user cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) is not a match for permission 'UserPermission(type=ldapGroup, ldapProfileID=default, ldapQuery=null, ldapBase=cn=ADM,ou=Touk,ou=Group,dc=touk,dc=pl)' (0ms) [172.17.0.1]
2020-11-25T18:34:14Z, DEBUG, ldap.UserInfoReader, {hIv2C} Helpdesk has no matching profiles for user cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) [172.17.0.1]
2020-11-25T18:34:14Z, TRACE, permission.UserPermissionUtility, {hIv2C} begin check for ldapQuery match for cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) using queryMatch: (objectClass=toukPerson) [172.17.0.1]
2020-11-25T18:34:14Z, TRACE, permission.UserPermissionUtility, {hIv2C} checking ldap to see if cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) matches '(objectClass=toukPerson)' [172.17.0.1]
2020-11-25T18:34:14Z, DEBUG, permission.UserPermissionUtility, {hIv2C} user cn=Michał Sołtys,ou=Touki,ou=People,dc=touk,dc=pl (default) is a match for permission 'UserPermission(type=ldapQuery, ldapProfileID=all, ldapQuery=(objectClass=toukPerson), ldapBase=ou=Touki,ou=People,dc=touk,dc=pl)' (1ms) [172.17.0.1]
```

**Comment by zwets** [comment]
Confirming this issue. @msoltyspl do you recall the last known good snapshot? Will need to roll back to get to the admin settings.

**Comment by jrivard** [comment]
Should be fixed in commit 496bb64, please re-open if issue still occurs.
